lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -88,11 +88,8 @@ private function _constructPdoDsn(array $params)
 
         if (isset($params['dbname'])) {
             $dsn .= 'dbname=' . $params['dbname'] . ' ';
-        } else {
-            // Used for temporary connections to allow operations like dropping the database currently connected to.
-            // Connecting without an explicit database does not work, therefore "postgres" database is used
-            // as it is certainly present in every server setup. 
-            $dsn .= 'dbname=postgres' . ' ';
+        } elseif (isset($params['default_dbname'])) {
+            $dsn .= 'dbname=' . $params['default_dbname'] . ' ';
```

Comment:
@deeky666 @kimhemsoe Ah yes, I was wrong - you can verify this using:

PDO
===

```
php -r "$pdo = new PDO('pgsql:host=localhost port=5432', 'your_username', 'your_password');"
```

If no db matching your username exists, you get:

```
PDOException: SQLSTATE[08006] [7] FATAL:  database "your_username" does not exist in Command line code on line 1

Call Stack:
    0.0000     125096   1. {main}() Command line code:0
    0.0000     125520   2. PDO->__construct() Command line code:1
```

pgsql
=====

```
php -r "$conn = pg_connect('host=localhost port=5432 user=your_username password=your_password');"
```

If no db matching your username exists, you get:

```
Warning: pg_connect(): Unable to connect to PostgreSQL server: FATAL:  database "your_username" does not exist in Command line code on line 1

Call Stack:
    0.0000     124792   1. {main}() Command line code:0
    0.0000     124928   2. pg_connect() Command line code:1
```